internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -342,8 +344,13 @@ func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string
 
 	watcherExecutable := selectWatcherExecutable(paths.Top(), previous, current)
 
+	rollbackWindow := u.upgradeSettings.Rollback.Window
+	if !isRollbackWindowFeatureAvailable(parsedVersion) {
```

Comment:
Maybe add a small comment here as to why we check the new agent version before passing a rollback window argument

Reply:
Added in 39ac19916c.